### DISCUSSION TITLE
Nested component save functionality

### DISCUSSION THREAD

**Discussion by ju5t** [livewire/livewire#2467] [4. General]
This is 100% meant as a discussion. I'm not necessarily looking for help, although it is a challenge I have yet to find a good solution for. I am wondering how others are doing this.

We have the following case:
- A customer create/edit component.
- It should manage multiple contact addresses (general, billing, shipping)
- It should manage multiple users.

This would quickly grow into a large component with all sorts of rules and methods to deal with all the functionality.

I quickly tend to look at nested components at that point. I think it is much prettier. It's a clean separation of functionality and it can be reused. Managing addresses and users would be needed in both the administrative and customer-facing part of a website. 

I did have a solution for this where I tried to tie them together with events. It wasn't great, but it was at least something. Here's how it works:

- Dispatch an event from the `save`-button.
- Listen for the event in both the parent as nested components.
- Save the component, send another event that notifies the notification AlpineJS component that we're finished.
- Have that AlpineJS component debounce for a few seconds and assume everything is saved if nothing comes in.

There are lots of problems with this:

- What if the validation in one of the nested components fail?
- What if a nested didn't actually save? The debounce will eventually message things are finished.

And so on.

It would be better to have a proper link - but the only way I can think of doing this is using traits. This would still need some sort of logic in the parent component but the traits can do the bulk of the work. This would be sort of clean and reusable, and it would allow you to link more advanced things together.

I wonder how others are doing this. I hope some of you can share some thoughts on this.

**Comment by joshhanley** [comment]
@ju5t are you on the Livewire discord? If so DM me @joshhanley#6022 have something that might interest you, that's not quite ready for sharing just yet.
